script/DeployMigrationStakeManager.s.sol

```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.19 <=0.9.0;
+
+import { BaseScript } from "./Base.s.sol";
+import { StakeManager } from "../contracts/StakeManager.sol";
+
+contract DeployMigrationStakeManager is BaseScript {
+    error DeployMigrationStakeManager_InvalidStakeTokenAddress();
+
+    address public prevStakeManager;
+
+    address public stakeToken;
+
+    constructor(address _prevStakeManager, address _stakeToken) {
+        prevStakeManager = _prevStakeManager;
+        stakeToken = _stakeToken;
+    }
+
+    function run() public returns (StakeManager) {
+        prevStakeManager = vm.envOr({ name: "PREV_STAKE_MANAGER", defaultValue: prevStakeManager });
+        stakeToken = vm.envOr({ name: "STAKE_TOKEN_ADDRESS", defaultValue: stakeToken });
+
+        if (stakeToken == address(0)) {
+            revert DeployMigrationStakeManager_InvalidStakeTokenAddress();
+        }
+
+        vm.startBroadcast(broadcaster);
+        StakeManager stakeManager = new StakeManager(stakeToken, prevStakeManager);
```

Comment:
Another thing I've just realised, it probably makes sense to extend this so that it automatically activates migration in the old manager.

If we make this part of this script, however, it's probably less usable in the testing context (which I think would be fine, as in tests we can always just do `new StakeManager()` individually.

Let me know what you think.